agent/api/container/container.go

```diff
@@ -109,15 +106,25 @@ type HealthStatus struct {
 	Output string `json:"output,omitempty"`
 }
 
-type ManagedAgent struct {
+type ManagedAgentState struct {
+	// ID of this managed agent state
+	ID string `json:"id:omitempty"`
 	// Status is the managed agent health status
 	Status apicontainerstatus.ManagedAgentStatus `json:"status:omitempty"`
-	// Name is always executeCommandAgent (until a new agent type is introduced)
-	Name string `json:"name,omitempty"`
 	// Reason is a placeholder for failure messaging
 	Reason string `json:"reason,omitempty"`
 	// LastStartedAt is the timestamp when the status last went from PENDING->RUNNING
 	LastStartedAt time.Time `json:"lastStartedAt,omitempty"`
+	// Metadata holds metadata about the managed agent
+	Metadata map[string]interface{} `json:"metadata,omitempty"`
+}
+
+type ManagedAgent struct {
+	ManagedAgentState
+	// Name is the name of this managed agent. This name is streamed down from ACS.
+	Name string `json:"name,omitempty"`
+	// Properties of this managed agent. Properties are streamed down from ACS.
+	Properties map[string]string `json:"properties,omitempty"`
```

Comment:
this can be a struct too. we don't have to keep it the same as ACS

Reply:
I prefer to keep it the same, otherwise we would need to build custom logic in `TaskFromACS` and remember to update it/add to it when the ACS model is modified.

Reply:
also makes sense to keep it generic, since its the managed agent;s struct